### PR TITLE
[mobile_static] Allow mcs/class tests to run to completion

### PR DESCRIPTION
**Pull request by alexanderkyte**
This fixes all problems with building the test .dll files for mobile_static and with test crashes related to build / preprocessor usage.

Note that the mono/tests tests are still broken here, so the jenkins script will mark those as failing.

### CODE REVIEW

**Comment by xmcclure** [comment]
This looks good to me

File: mcs/build/rules.make [hunk 1]
```diff
@@ -202,16 +202,21 @@ STD_TARGETS = test run-test run-test-ondotnet clean install uninstall doc-update
...
 	$(MAKE) do-all TOP_LEVEL_DO=do-all
 	$(MAKE) aot-all-profile
 
-aot-all-profile:
-	$(Q_AOT) cd $(topdir)/class/lib/$(PROFILE)/ && $(COMPILE_ALL_PROFILE_ASSEMBLIES) &> $(PROFILE)-aot.log
+# This can run in parallel
+.PHONY: aot-all-profile
+aot-all-profile: $(patsubst %,$(topdir)/class/lib/$(PROFILE)/%.dylib,$(AOT_PROFILE_ASSEMBLIES))
+
+$(topdir)/class/lib/$(PROFILE)/%.dylib: $(topdir)/class/lib/$(PROFILE)/%
```
Comment (akoeplinger) [inline mcs/build/rules.make]: do we want a .so target as well or do we only care about OSX for now?

Comment (akoeplinger) [inline mcs/build/rules.make]: or just use PLATFORM_AOT_SUFFIX

File: mcs/build/rules.make [hunk 2]
```diff
@@ -202,16 +202,21 @@ STD_TARGETS = test run-test run-test-ondotnet clean install uninstall doc-update
...
+	@ mkdir -p $(topdir)/class/lib/$(PROFILE)/$*_bitcode
```
Comment (akoeplinger) [inline mcs/build/rules.make]: maybe append _temp or something so it's clear this shouldn't stay around

File: mcs/class/System/Makefile [hunk 3]
```diff
@@ -19,7 +19,13 @@ TEST_RESOURCES = \
 	Test/System/test-uri-props-manual.txt \
 	Test/System/test-uri-relative-props.txt
 
+TEST_LIB_REFS = Mono.Security System.Data System.Xml System.Core
```
Comment (akoeplinger) [inline mcs/class/System/Makefile]: this seems redundant as we define it right below anyway

**Comment by akoeplinger** [comment]
LGTM apart from the comments I made

**Comment by alexanderkyte** [comment]
The only failure is in the runtime suite, and that shouldn't be changed by mcs/class changes. Also, I've addressed your comments @akoeplinger .

**Pull request merged**